### PR TITLE
Using new package naming without spaces in the installers

### PR DESCRIPTION
**Pull request by evtrados**
This request came from @mahtDFR and marketing in general, since so far the Cura installers (especially the ones for Windows) had spaces, i.e., "Ultimaker Cura-4.6.0-win64.exe" (or .msi). This PR started as a tryout fix for that, but soon it was realized that the CPack generators are a bit more complicated than that and, as a result, the initial investigation and fix resulted in a slightly bigger update than initially intended. Things that have been changed:
- CPACK_PACKAGE_NAME. This variable is now set to something like "Ultimaker Cura X.Y". That way the WIX CPack generator will create nice uninstall links in the Windows start menu folder (i.e., C:\ProgramData\Microsoft\Windows\Start Menu\Programs).
- Creating a versioned uninstaller in the start menu also when using the NSIS CPack Generator.
- Whenever set, adding the build type of Cura (e.g., testing) in the eventual Windows installer, shortcut and installation directory.

All this is part of CURA-7430

### CODE REVIEW

File: packaging/cpackconfig_common.cmake [hunk 1]
```diff
@@ -1,13 +1,19 @@
 cpack_add_component(_cura
-                    DISPLAY_NAME "Ultimaker Cura"
-                    Description "Ultimaker Cura Executable and Data Files"
-                    REQUIRED
+		DISPLAY_NAME "Ultimaker Cura ${CURA_BUILDTYPE}"
```
Comment (ninovanhooff) [inline packaging/cpackconfig_common.cmake:2]: Will a regular build be called "Ultimaker Cura " (note the extra space)?

File: packaging/cpackconfig_nsis.cmake [hunk 2]
```diff
@@ -11,6 +11,7 @@ cpack_add_component(arduino DISPLAY_NAME "Arduino Drivers")
 # ========================================
 set(CPACK_NSIS_ENABLE_UNINSTALL_BEFORE_INSTALL ON)
 set(CPACK_NSIS_EXECUTABLES_DIRECTORY ".")
+set(CPACK_NSIS_STARTMENU_DIRECTORY "Ultimaker Cura")
```
Comment (ninovanhooff) [inline packaging/cpackconfig_nsis.cmake:14]: Why not distinguish between Cura and Enterprise here? Is NSIS community edition only?

File: packaging/cpackconfig_common.cmake (hunk 1 above)
Comment (evtrados) [reply packaging/cpackconfig_common.cmake:2]: I was a bit hesitant about that one, too. The only place the DISPLAY_NAME is used is during installation, where the user reads the license, etc., in cases like: "Installing Ultimaker Cura " or "Ultimaker Cura  Setup" (2 spaces in the last example). After checking it though, it is not really obvious and easily visible. There is a workaround of course, if you don't like it (and I agree it's not the "best" way to do it).

File: packaging/cpackconfig_nsis.cmake (hunk 2 above)
Comment (evtrados) [reply packaging/cpackconfig_nsis.cmake:14]: Indeed, NSIS creates .exe installers, which will never be used for Enterprise builds.

File: packaging/cpackconfig_common.cmake (hunk 1 above)
Comment (ninovanhooff) [reply packaging/cpackconfig_common.cmake:2]: If it's only visible there i'll accept it

File: packaging/cpackconfig_common.cmake [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+		Description "Ultimaker Cura ${CURA_BUILDTYPE} Executable and Data Files"
+		REQUIRED
```
Comment (Ghostkeeper) [inline packaging/cpackconfig_common.cmake:4]: Mind that this file is using a mix of tabs and spaces now. Our code style conventions dictate to indent with 4 spaces.

Comment (evtrados) [reply packaging/cpackconfig_common.cmake]: Sorry @Ghostkeeper, I was not aware of that. I replaced the tabs with 4 spaces as indicated, It should be fine now :)

**Pull request merged**